libs/editor/egui_editor/src/input/canonical.rs

```diff
@@ -251,18 +251,18 @@ pub fn calc(
         Event::PointerButton { pos, button: PointerButton::Primary, pressed: true, modifiers }
             if click_checker.ui(*pos) =>
         {
-            pointer_state.press(now, *pos, *modifiers);
+            pointer_state.press(now, *pos, click_checker.pos_to_char_offset(*pos), *modifiers);
```

Comment:
the fix is to convert the pointer position to the character offset when the pointer is pressed instead of when it's dragged/released